services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -110,38 +112,52 @@ async def create_computation(
                 detail=f"Projet {job.project_id} already started, current state is {pipeline_state}",
             )
 
-        # create the DAG
+        # create the computational DAG
         dag_graph = create_dag_graph(project.workbench)
         # validate DAG
         if not nx.is_directed_acyclic_graph(dag_graph):
             raise HTTPException(
                 status_code=status.HTTP_400_BAD_REQUEST,
                 detail=f"Project {job.project_id} is not a valid directed acyclic graph!",
             )
-        # find the entrypoints
-        entrypoints = find_entrypoints(dag_graph)
-        if not entrypoints:
-            raise HTTPException(
-                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
-                detail=f"Project {job.project_id} has no services to compute",
-            )
-        # FIXME: directly pass the tasks to celery instead of the current recursive way
-        # ok so publish the tasks
-        await computation_pipelines.publish_pipeline(project.uuid, dag_graph)
-        await computation_tasks.publish_tasks_from_project(project, director_client)
-        # trigger celery
-        task = celery_client.send_computation_task(job.user_id, job.project_id)
-        background_tasks.add_task(background_on_message, task)
-        log.debug(
-            "Started computational task %s for user %s based on project %s",
-            task.id,
-            job.user_id,
-            job.project_id,
+
+        if job.start_pipeline:
+            # find the entrypoints, if not the pipeline cannot be started
+            entrypoints = find_entrypoints(dag_graph)
+            if not entrypoints:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f"Project {job.project_id} has no services to compute",
+                )
+
+        # ok so put the tasks in the db
+        await computation_pipelines.upsert_pipeline(
+            project.uuid, dag_graph, job.start_pipeline
         )
+        await computation_tasks.upsert_tasks_from_project(
+            project, director_client, job.start_pipeline
+        )
+
+        if job.start_pipeline:
```

Comment:
Suggestion: Could `job` be called `comp_task_create` to reflect the type or rename the type into something like `job_start` and add some comments on the type?

Reply:
no I will probably change it to job in the near future. we start jobs...